OpenRobertaServer/src/main/java/de/fhg/iais/roberta/util/archiver/ProgramsXmlArchiver.java

```diff
@@ -27,12 +28,12 @@ public ProgramsXmlArchiver(int ownerId) {
 
     private void openStreams() {
         baos = new ByteArrayOutputStream();
-        zos = new ZipOutputStream(baos);
+        zipOutputStream = new ZipOutputStream(baos);
     }
 
-    public void closeStreams() throws Exception {
+    public void closeStreams() throws IOException  {
         baos.close();
```

Comment:
an exception may leave a stream open and leak resources. Better use a helper like:
```
private static void closeQuietly(AutoCloseable statement) {
        if ( statement != null ) {
            try {
                statement.close();
            } catch ( Exception e ) {
                // OK
            }
        }
    }
```